sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JsonFileFormat.scala

```diff
@@ -97,46 +91,13 @@ class JsonFileFormat extends TextBasedFileFormat with DataSourceRegister {
     val broadcastedHadoopConf =
       sparkSession.sparkContext.broadcast(new SerializableConfiguration(hadoopConf))
 
-    val parsedOptions: JSONOptions = new JSONOptions(options)
-    val columnNameOfCorruptRecord = parsedOptions.columnNameOfCorruptRecord
-      .getOrElse(sparkSession.sessionState.conf.columnNameOfCorruptRecord)
+    val parsedOptions = new JSONOptions(options,
+      sparkSession.sessionState.conf.columnNameOfCorruptRecord)
+    val readFile = JsonDataSource(parsedOptions).readFile _
```

Comment:
this closure has the reference of the outer pointer, so we will still broadcast the `JsonDataSource`. how about
```
val columnNameOfCorruptRecord = sparkSession.sessionState.conf.columnNameOfCorruptRecord)

(file: PartitionedFile) => {
  val parsedOptions  =...
  val parser = new JacksonParser(requiredSchema, parsedOptions)
  JsonDataSource(parsedOptions).readFile...
}
```